bevy_lint/src/lints/insert_event_resource.rs

```diff
@@ -0,0 +1,226 @@
+//! Checks for the `Events<T>` resource being manually inserted through `App::init_resource()` or
+//! `App::insert_resource()` instead of with `App::add_event()`.
+//!
+//! # Why is this bad?
```

Comment:
We should replace "why is this bad" with "why we lint for this".

reasoning: "bad" is a value judgement which can be internalized by beginners (about their code, skills, and themselves). We can replace this with something that is both more specific to what we mean ("Why we lint for this") and doesn't have these connotations.

```suggestion
//! # Why we lint for this
```

Reply:
I would personally leave it as is:
- this mirrors [clippy](https://rust-lang.github.io/rust-clippy/master/index.html#/too_many_arguments)'s page
- This *is* a value judgement, and we shouldn't hide it.
- imo it's okay to say that a pattern a user is using is bad. I think the writing style is impersonal enough to make it clear that this is mot about personal quality. 

But then again, I am not the target group that this is supposed to protect. If someone comes up and tells me that clippy's value judgement made them feel inadequate (beyond "ah, here's how I can do better, that's good to know"), I will change my stance accordingly.

Reply:
Actually, I'm interested in @alice-i-cecile's opinion, since the linter should reflect Bevy's communication style. If Bevy at large avoids "bad", we should do that too.

Reply:
I heavily prefer "why we lint for this" or "Motivation". It's less judgemental, and better accounts for style or conditional lints. I wish clippy would change too!

Reply:
Consider my stance changed then :)